dapp/src/components/wrap/WrapHomepage.js

```diff
@@ -0,0 +1,308 @@
+  const [needsApproval, setNeedsApproval] = useState()
+  const [rate, setRate] = useState()
+
+  const account = useStoreState(AccountStore, (s) => s.address)
+  const web3react = useWeb3React()
+  const { library } = web3react
+
+  const { ousd, wousd } = useStoreState(ContractStore, (s) => s.contracts)
+
+  const signer = (contract) => {
+    return contract.connect(library.getSigner(account))
+  }
+
+  useEffect(() => {
+    if (!wousd) {
+      return
+    }
+    const wrapEstimate = async () => {
+      let estimate
+      if (!inputAmount) {
+        estimate = 0
+      } else if (swapMode === 'mint') {
+        estimate = await displayCurrency(
+          await wousd.convertToShares(
+            calculateSwapAmounts(inputAmount, 18).swapAmount
+          ),
+          wousd
+        )
+      } else {
+        estimate = await displayCurrency(
+          await wousd.convertToAssets(
+            calculateSwapAmounts(inputAmount, 18).swapAmount
+          ),
+          ousd
+        )
+      }
+      setWrapEstimate(estimate)
+    }
+    const approvalNeeded = () => {
+      if (!allowancesLoaded) {
+        return
+      }
+      setNeedsApproval(
+        parseFloat(allowances.ousd.wousd) < inputAmount ? 'wousd' : ''
+      )
+    }
+    const calculateRate = async () => {
+      const conversionRate = await displayCurrency(
+        await wousd.convertToAssets(calculateSwapAmounts(1, 18).swapAmount),
+        wousd
+      )
+      setRate(conversionRate)
+    }
+    wrapEstimate()
+    approvalNeeded()
+    calculateRate()
+  }, [inputAmount, wousd, allowances])
+
+  useEffect(() => {
+    // currencies flipped
+    if (previousSwapMode !== swapMode) {
+      localStorage.setItem(lastSelectedSwapModeKey, swapMode)
+      const otherCoinAmount = Math.floor(wrapEstimate * 1000000) / 1000000
+      setInputAmount(Math.floor(otherCoinAmount * 100) / 100)
+    }
+  }, [swapMode])
+
+  const errorMap = [
+    {
+      errorCheck: (err) => {
+        return err.name === 'EthAppPleaseEnableContractData'
+      },
+      friendlyMessage: fbt(
+        'Contract data not enabled. Go to Ethereum app Settings and set "Contract Data" to "Allowed"',
+        'Enable contract data'
+      ),
+    },
+    {
+      errorCheck: (err) => {
+        return err.message.includes(
+          'Failed to sign with Ledger device: U2F DEVICE_INELIGIBL'
+        )
+      },
+      friendlyMessage: fbt(
+        'Can not detect ledger device. Please make sure your Ledger is unlocked and Ethereum App is opened.',
+        'See ledger connected'
+      ),
+    },
+  ]
+
+  const onMintingError = (error) => {
+    if (errorMap.filter((eMap) => eMap.errorCheck(error)).length > 0) {
+      setBuyErrorToDisplay(error)
+    }
+  }
+
+  const swapMetadata = () => {
+    const coinGiven = swapMode === 'mint' ? 'wousd' : 'ousd'
+    const coinReceived = swapMode === 'mint' ? 'ousd' : 'wousd'
+    const swapAmount = swapMode === 'mint' ? wrapEstimate : inputAmount
```

Comment:
Hey tanks for noticing this. I thing it will be easier to measure stuff in OUSD from the analytics point of view. Have switched it around in this commit: https://github.com/OriginProtocol/origin-dollar/pull/988/commits/f1b913b46ceb93f08a5443e78309f15242a236a3